gax-httpjson/src/main/java/com/google/api/gax/httpjson/InstantiatingHttpJsonChannelProvider.java

```diff
@@ -193,33 +197,39 @@ public static Builder newBuilder() {
   }
 
   public static final class Builder {
-    private ExecutorProvider executorProvider;
+    private Executor executor;
     private HeaderProvider headerProvider;
     private String endpoint;
     private HttpTransport httpTransport;
 
     private Builder() {}
 
     private Builder(InstantiatingHttpJsonChannelProvider provider) {
-      this.executorProvider = provider.executorProvider;
+      this.executor = provider.executor;
       this.headerProvider = provider.headerProvider;
       this.endpoint = provider.endpoint;
       this.httpTransport = provider.httpTransport;
     }
 
     /**
-     * Sets the ExecutorProvider for this TransportChannelProvider.
+     * Sets the Executor for this TransportChannelProvider.
      *
      * <p>This is optional; if it is not provided, needsExecutor() will return true, meaning that an
      * Executor must be provided when getChannel is called on the constructed
      * TransportChannelProvider instance. Note: InstantiatingHttpJsonChannelProvider will
      * automatically provide its own Executor in this circumstance when it calls getChannel.
      */
-    public Builder setExecutorProvider(ExecutorProvider executorProvider) {
-      this.executorProvider = executorProvider;
+    public Builder setExecutor(Executor executor) {
+      this.executor = executor;
       return this;
     }
 
+    /** @deprecated. Please use {@link #setExecutor(Executor)}. */
+    @Deprecated
+    public Builder setExecutor(ScheduledExecutorService executor) {
```

Comment:
This should be called `setExecutorProvider` and accept ExecutorProvider as an argument (i.e. put back the old method) and deprecate it

Reply:
Done.